pkg/ccl/testccl/sqlccl/sqlccl_test: TestExplainGist failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestExplainGist [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16570972?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16570972?buildTab=artifacts#/) on master @ [c01ba357832133f0393fc8d557afe9b31e6da768](https://github.com/cockroachdb/cockroach/commits/c01ba357832133f0393fc8d557afe9b31e6da768):


```
=== RUN   TestExplainGist
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/17d4c94e97ab6a70b4ba9df15e5f218d/logTestExplainGist523061330
    test_log_scope.go:81: use -show-logs to present logs inline
    explain_test.go:310: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/17d4c94e97ab6a70b4ba9df15e5f218d/logTestExplainGist523061330
--- FAIL: TestExplainGist (9.23s)
=== RUN   TestExplainGist/main
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    explain_test.go:257: pq: descriptor is being dropped
    --- FAIL: TestExplainGist/main (6.11s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainGist.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41546

--- a/pkg/ccl/testccl/sqlccl/BUILD.bazel
+++ b/pkg/ccl/testccl/sqlccl/BUILD.bazel
@@ -39,7 +39,6 @@ go_test(
         "//pkg/settings/cluster",
         "//pkg/spanconfig",
         "//pkg/sql",
-        "//pkg/sql/catalog",
         "//pkg/sql/gcjob",
         "//pkg/sql/isql",
         "//pkg/sql/lexbase",

--- a/pkg/ccl/testccl/sqlccl/explain_test.go
+++ b/pkg/ccl/testccl/sqlccl/explain_test.go
@@ -16,10 +16,10 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/internal/sqlsmith"
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/lexbase"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltestutils"
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -253,7 +253,7 @@ func TestExplainGist(t *testing.T) {
 			if err != nil {
 				// We might be still in the process of cancelling the previous
 				// DROP operation - ignore this particular error.
-				if !errors.Is(err, catalog.ErrDescriptorDropped) {
+				if !testutils.IsError(err, "descriptor is being dropped") {
 					t.Fatal(err)
 				}
 				continue

